File: wiki_java/src/main/java/com/github/klee0kai/wiki/provide/binding/App2.java
package com.github.klee0kai.wiki.provide.binding;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.test.mowgli.galaxy.Sun;

public class App2 {

    public static void main(String[] args) {
        SunSystemComponent DI = Stone.createComponent(SunSystemComponent.class);
        Sun sun = new Sun();
        DI.sun(sun);


        Sun sunFromDI = DI.sun(null);
        System.out.println("sunFromDI " + sunFromDI);
    }

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/binding/App2.kt
package com.github.klee0kai.wiki.provide.binding

import com.github.klee0kai.test.mowgli.galaxy.Sun

fun main(args: Array<String>) {
    val DI = SunSystemComponentStoneComponent()
    val sun = Sun()
    DI.sun(sun)


    val sunFromDI = DI.sun(null)
    println("sunFromDI $sunFromDI")
}
